MantidQt/MantidWidgets/src/DataProcessorUI/GenericDataProcessorPresenter.cpp

```diff
@@ -600,25 +600,26 @@ GenericDataProcessorPresenter::reduceRow(const std::vector<std::string> &data) {
     // The column's name
     auto columnName = m_whitelist.colNameFromColIndex(i);
 
-    if (m_preprocessMap.count(columnName)) {
+    if (m_preprocessMap.count(columnName) && globalOptions.count(columnName)) {
       // This column needs pre-processing
 
       const std::string runStr = data.at(i);
 
       if (!runStr.empty()) {
 
-        auto preprocessor = m_preprocessMap[columnName];
+        auto preprocessor = m_preprocessMap.at(columnName);
 
         // Global pre-processing options for this algorithm as a string
-        const std::string options = globalOptions[columnName];
+        const std::string options = globalOptions.at(columnName);
 
         auto optionsMap = parseKeyValueString(options);
         auto runWS = prepareRunWorkspace(runStr, preprocessor, optionsMap);
+
         alg->setProperty(propertyName, runWS->getName());
       }
     } else {
       // No pre-processing needed
-      auto propertyValue = data.at(i);
+      const std::string propertyValue = data.at(i);
```

Comment:
I think I was trying to do what you have done (pass an empty string as option). I believe your solution works better as it does not involve having to potentially change any tests though, so I will try to implement that.